### ISSUE TITLE
Preserving unpushed commits

### ISSUE THREAD

**Issue by simonwheatley** [Automattic/vip-go-quickstart#23]
Currently everytime `./qs-init.sh` is run, the repo is removed and re-cloned. If this command was run when there were unpushed commits, those commits would be lost in time, like tears… in… rain.

**Comment by simonwheatley** [comment]
Possible solution:

Have an `archive` folder, for repos, then when the QS is (re-)init'ed:

1. Move any existing `go-client-repo` to `archive`, naming it as per the client slug
2. Check the `archive` storage for any repo with the new client slug, move if it found (assumption: this archived repo refers to the same Git repo as the requested repo, although I guess we could check that)
3. If no archived repo found, clone the requested archive as new

The dev is then responsible for ensuring they `fetch`, `pull`, `checkout` the code they need to work on.

Moves are pretty quick, certainly quicker than cloning.

**Comment by ethitter** [comment]
I agree that this experience needs to be better; related, #21. As I've been thinking about this problem, an archive folder like you've described seems the best option. We're really dealing with two similar but different issues (preserving data, be it inside the VM or on the host machine), but a common approach will make this more manageable going forward.

I hadn't thought to key the archive by git repo, but that makes a lot more sense than relying on client slug. If we prompt the user to either reuse an archive or clone anew, we can also address the dev's responsibility to keep the local copy up-to-date by adding some warning during init, and we don't need to worry about branches (related, #27). Doing so also aids in an emergency, when I might just need to start fresh to find out what broke what.

For uploads, I don't see any reason to move them to the host machine unless we're also going to export the database, since nothing's currently preserved when `vagrant destroy` runs anyway. Keying this archive is less straightforward, though, and probably leaves us with inconsistency. The database uses the client slug to maintain separation between inits, so it would make sense to do the same for uploads. If we do that, though, the difference between host and VM archive keying could get annoying. Addressing this makes a further case for a prompt during init, which we could use to then discard any VM-side archives just as we would the host-side git clone. At the same time, if the user reuses a client slug for unrelated projects, the mixed keying could lead to deleting the wrong data. Nonetheless, this seems a minor concern, and something we can warn about. It's always possible I'm overthinking this. :)

**Comment by simonwheatley** [comment]
@ethitter I'm leaning towards this being part of the [MVP Milestone](https://github.com/Automattic/vip-go-quickstart/milestones/Minimum%20Viable%20Product), what do you think?
